src/auth/user-record.ts

```diff
@@ -66,16 +66,19 @@ export interface CreateRequest extends UpdateRequest {
  * @constructor
  */
 export class UserMetadata {
-  public readonly creationTime: string;
-  public readonly lastSignInTime: string;
+  public readonly creationTime?: string | null;
```

Comment:
The `?` isn't strictly necessary as `parseDate` always returns `null` if the input date is undefined. But I think that's more implementation detail-ish, so I've left open the possibility of undef here.

Reply:
These are not currently defined as optional (both here and in the d.ts file). So may be we can keep it as it was?

Reply:
The problem is that these reflect the proto, so when the values come back from the server, there isn't any guarantee that these will actually be populated. (For instance, the lastSignInTime will not be present if the user has never logged in.) As it happens, our implementation will notice that and set lastSignInTime to null. I'm not sure if that happens for all of our objects or not... though we could explicitly set things to null in that situation should there be cases where things are otherwise left out, thereby dropping the '?'.

But not adding the '| null' isn't great since it could legitimately be null (or missing). We just didn't notice before because the property was being set indirectly. We could detect this and use a sentinel value instead (empty string in this case?). To be clear, in some circumstances, we are setting these non-nullable values to null today; typescript just isn't noticing.

In other words, I suspect many instances of this throughout the code base are legitimately optional and/or nullable, but because they've been set indirectly, we haven't noticed.

Reply:
Marking is nullable is fine (and is the right thing to do, since we explicitly initialize it to null). But marking it optional (`?`) seems wrong to me, since it's never undefined.

Reply:
Ah, I understand you now. I've removed the '?'.

Note that other classes (that I haven't made this conversion to yet) may not explicitly set to null and instead just not set the property. (Possibly; haven't checked yet.) I'll set the types to reflect what they are (like we've now done here) and we can figure out then if there's an inconsistency issue that needs addressing.